src/back/glsl/mod.rs

```diff
@@ -333,6 +333,18 @@ struct VaryingName<'a> {
 impl fmt::Display for VaryingName<'_> {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match *self.binding {
+            crate::Binding::Location {
+                index: Some(index), ..
+            } => {
+                let prefix = match (self.stage, self.output) {
+                    (ShaderStage::Compute, _)
+                    | (ShaderStage::Vertex, _)
+                    | (ShaderStage::Fragment, false) => unreachable!(),
+                    // fragment to pipeline
+                    (ShaderStage::Fragment, true) => "fs2p",
+                };
+                write!(f, "_{prefix}_location{index}",)
```

Comment:
nitpick: Fun fact, `rustfmt` doesn't remove trailing commas from macros, even when not needed for macros like `write!(…)`, because nobody's done the work of special-casing for "known" macros where it's okay yet: https://github.com/rust-lang/rustfmt/issues/3065